lib/spack/spack/test/installer.py

```diff
@@ -231,32 +231,12 @@ def _spec(spec, preferred_mirrors=None):
 
 def test_try_install_from_binary_cache(install_mockery, mock_packages,
                                        monkeypatch):
-    """Tests SystemExit path for_try_install_from_binary_cache.
-
-       This test does not make sense.  We tell spack there is a mirror
-       with a binary for this spec and then expect it to die because there
-       are no mirrors configured."""
-    # def _mirrors_for_spec(spec, full_hash_match=False):
-    #     spec = spack.spec.Spec('mpi').concretized()
-    #     return [{
-    #         'mirror_url': 'notused',
-    #         'spec': spec,
-    #     }]
-
+    """Test return false when no match exists in the mirror"""
```

Comment:
I think changes in this file can be extracted in their own small PR and merged straight away

Reply:
Ok, extracted these changes and created #29305.

Reply:
Merged